src/main/java/org/apache/commons/text/numbers/DoubleFormat.java

```diff
@@ -0,0 +1,736 @@
+            this.plainFormatMaxDecimalExponent = plainFormatMaxDecimalExponent;
+            return this;
+        }
+
+        /** Set the minimum decimal exponent for numbers formatted as plain decimal strings when
+         * using the {@link DoubleFormat#MIXED MIXED} format type. If the number being formatted
+         * has an absolute value less than <code>10<sup>plainFormatMaxDecimalExponent + 1</sup></code> and
+         * greater than or equal to <code>10<sup>plainFormatMinDecimalExponent</sup></code> after any
+         * necessary rounding, then the formatted result will use the {@link DoubleFormat#PLAIN PLAIN} format type.
+         * Otherwise, {@link DoubleFormat#SCIENTIFIC SCIENTIFIC} format will be used. For example,
+         * if this value is set to {@code -2}, the number {@code 0.01} will be formatted as {@code "0.01"}
+         * while {@code 0.0099} will be formatted as {@code "9.9E-3"}.
+         *
+         * <p>The default value is {@value #DEFAULT_PLAIN_FORMAT_MIN_DECIMAL_EXPONENT}.
+         *
+         * <p>This value is ignored for formats other than {@link DoubleFormat#MIXED}.
+         * @param plainFormatMinDecimalExponent maximum decimal exponent for values formatted as plain
+         *      strings when using the {@link DoubleFormat#MIXED MIXED} format type.
+         * @return this instance
+         * @see #plainFormatMinDecimalExponent(int)
+         */
+        public Builder plainFormatMinDecimalExponent(final int plainFormatMinDecimalExponent) {
+            this.plainFormatMinDecimalExponent = plainFormatMinDecimalExponent;
+            return this;
+        }
+
+        /** Set the flag determining whether or not the zero string may be returned with the minus
+         * sign or if it will always be returned in the positive form. For example, if set to true,
+         * the string {@code "-0.0"} may be returned for some input numbers. If false, only {@code "0.0"}
+         * will be returned, regardless of the sign of the input number. The default value is {@code true}.
+         * @param signedZero if true, the zero string may be returned with a preceding minus sign; if false,
+         *      the zero string will only be returned in its positive form
+         * @return this instance
+         */
+        public Builder allowSignedZero(final boolean signedZero) {
+            this.signedZero = signedZero;
+            return this;
+        }
+
+        /** Set the string containing the digit characters 0-9, in that order. The
+         * default value is the string {@code "0123456789"}.
+         * @param digits string containing the digit characters 0-9
+         * @return this instance
+         * @throws NullPointerException if the argument is null
+         * @throws IllegalArgumentException if the argument does not have a length of exactly 10
+         */
+        public Builder digits(final String digits) {
+            Objects.requireNonNull(digits, "Digits string cannot be null");
```

Comment:
`digits` appears to be null-safe. But `minusSign` and other builder methods are not. Was it intentional? Or should we follow the example of this method and protect others too?

e.g.

```java
        DoubleFunction<String> fmt = DoubleFormat.PLAIN.builder()
                .allowSignedZero(true)
                .minusSign((Character) null) // NPE
                .build();
        String s = fmt.apply(1.3);
        System.out.println(s);
```

Reply:
I don't think this needs any changes since `minusSign` and others only accept primitive values. The NPE in the example above would be thrown when computing the arguments to pass to `minusSign`.